### ISSUE TITLE
A question about the initialization of biaffine module

### ISSUE THREAD

**Issue by speedcell4** [yzhangcs/parser#100]
Hi~

Why do you initialize the `weight` to be all zeros?

https://github.com/yzhangcs/parser/blob/16ad39534957bc4ee7af6ca8874de79332e8e8a2/supar/modules/affine.py#L54-L55

As I remember, PyTorch initializes the weight in [a different way](https://github.com/pytorch/pytorch/blob/ce700da7f4b3f2930a83f2f8995c0039c7838faf/torch/nn/modules/linear.py#L182-L186). Could you please explain your different choice?

**Comment by yzhangcs** [comment]
@speedcell4 
Hi~

My implementation references [this repo](https://github.com/jcyk/Dynet-Biaffine-dependency-parser/blob/master/models/baseParser.py#L49-L50).
In practice, I found the default initialization strategy or orthogonal initialization also work very well.

**Comment by speedcell4** [comment]
So there is no significant performance difference between these three initialization strategies, i.e., zeros, uniform, and orthogonal.
Honestly, using all zeros is quite strange to me, doesn't it map every input vector to a zero vector, and accumulate zero gradients?

**Comment by speedcell4** [comment]
Okay, I got it. Thanks for you replies~

**Comment by yzhangcs** [comment]
@speedcell4 
> doesn't it map every input vector to a zero vector, and accumulate zero gradients?

 I don't think this could lead to zero gradients as gold dependencies can back propagate **1** gradients to Biaffine layers and help get rid of zero init quickly. 
However, it does have some potential problems, and in some cases zero init is difficult to train.
In practice, I found normal init performs much better than zero init on Chinese Constituency Parsing.

**Comment by speedcell4** [comment]
I understand it now, thank you for your explanation~
